plugins/modules/nmcli.py

```diff
@@ -2602,6 +2611,7 @@ def main():
             downdelay=dict(type='int'),
             updelay=dict(type='int'),
             xmit_hash_policy=dict(type='str'),
+            fail_over_mac=dict(type='int'),
```

Comment:
Documentation such as:
https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/networking_guide/sec-Using_Channel_Bonding#s3-modules-bonding-directives

indicate that this parameter, in the CLI, can also receive textual values. I would suggest making it a `str` rather than an `int`. Param `xmit_hash_policy` above falls under the same category and does that.

Moreover, this parameter (and `xmit_hash_policy` for that matter) have a fixed list of accepted values, so I would suggest you consider using `choices` in its definition to validate those values.

Reply:
great suggestion, added choices and converted to type str. used the non integer values to stay consistent with the rest of the module